### PR TITLE
Fix dns resolution skipping over nameservers with valid responses

### PR DESCRIPTION
**Pull request by dwelch-r7**
Continuation of https://github.com/rapid7/metasploit-framework/pull/18660

Recommend turning whitespace changes off for this one https://github.com/rapid7/metasploit-framework/pull/18662/files?diff=unified&w=1

There have been a few issues reported with DNS resolution recently this PR aims to resolve (😁) those

Noticed a couple of issues:
1. Only a single nameserver was ever being pulled out of `/etc/resolv.conf` with the last nameserver overwriting the previous one(s)
2. Looping over the nameservers until we get a valid response was not correctly returning the answer from DNS instead moving on to the next nameserver until it ran out and saying the dns resolution was failing

### CODE REVIEW

File: lib/net/dns/resolver.rb [hunk 1]
```diff
@@ -1093,7 +1093,7 @@ def parse_config_file
             when /^\s*search\s+(.*)/
               self.searchlist = $1.split(" ")
             when /^\s*nameserver\s+(.*)/
-              self.nameservers = $1.split(" ")
+              self.nameservers += $1.split(" ")
             end
           end
         end
```
Comment (adfoster-r7) [inline lib/net/dns/resolver.rb:1099]: Does the windows codepath get handled correctly? Probably worth a check after the unix-y path works 👀

File: lib/rex/proto/dns/resolver.rb [hunk 2]
```diff
@@ -312,48 +312,43 @@ def send_tcp(packet, packet_data, nameservers, prox = @config[:proxies])
     # @return ans [String] Raw DNS reply
     def send_udp(packet,packet_data, nameservers)
       ans = nil
-      response = ""
       nameservers.each do |ns, socket_options|
-        catch(:next_ns) do
+        @config[:udp_timeout].timeout do
```
Comment (bwatters-r7) [inline lib/rex/proto/dns/resolver.rb:316]: Does this support timeout?

File: lib/net/dns/resolver.rb (hunk 1 above)
Comment (dwelch-r7) [reply lib/net/dns/resolver.rb:1099]: according to this random issue report (I couldn't find a better example of what gets returned) it _should_ be correct for windows
https://bugs.ruby-lang.org/issues/12604

but I'll be sure to verify this on windows myself too

File: lib/rex/proto/dns/resolver.rb (hunk 2 above)
Comment (dwelch-r7) [reply lib/rex/proto/dns/resolver.rb:316]: It sure does, I'll make the change

File: lib/net/dns/resolver.rb (hunk 1 above)
Comment (adfoster-r7) [reply lib/net/dns/resolver.rb:1099]: I have a windows dev env set up too, let me know if you've got replication steps you want me to follow :+1:

**Comment by adfoster-r7** [comment]
Can we verify with the docs/schema that there isn't more edegcases to consider in the /etc/resolv.conf config format? 🤞

**Comment by dwelch-r7** [comment]
> Can we verify with the docs/schema that there isn't more edegcases to consider in the /etc/resolv.conf config format? 🤞
> 
> Edit: Looks like there's some `options` that we could potentially support, but not needed in this PR

Forgot to reply to this earlier apologies
yea was looking through these docs, only thing that stands out like you said are the options 
https://www.ibm.com/docs/en/aix/7.3?topic=formats-resolvconf-file-format-tcpip

**Comment by adfoster-r7** [comment]
Not a blocker: This would be easier to test if there was a top level resolve command, similar to in Meterpreter 😄

**Comment by adfoster-r7** [comment]
Worked against my custom /etc/resolv.conf file :+1:

```
search foo.bar.com localdomain
nameserver x.x.x.x
nameserver x.x.x.x
```

**Comment by adfoster-r7** [comment]
## Release Notes

Fixes an edgecase where `features set dns_feature true` did not correctly parse a user's `/etc/resolv.conf` file if there were multiple nameservers present

**Comment by adfoster-r7** [comment]
> Not a blocker: This would be easier to test if there was a top level resolve command, similar to the one in Meterpreter 😄

I think @zeroSteiner has got us covered here now with the work he's doing in the same area🤞

**Pull request merged**